src/components/Projects/WBS/WBSDetail/AddTask/components/TagsSearch.jsx

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable function-paren-newline */
+/* eslint-disable implicit-arrow-linebreak */
+/* eslint-disable arrow-parens */
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-param-reassign */
+/* eslint-disable react/prop-types */
+/* eslint-disable max-len */
+/* eslint-disable react/no-array-index-key */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+import React, { useState } from 'react';
+import TagSent from './TagSent';
+import './TagsSearch.css';
+
+function TagsSearch({ data, placeholder }) {
+  const [tags, setTags] = useState([]);
+  const [isHidden, setIsHidden] = useState(false);
+  const [filteredData, setFilteredData] = useState([]);
+  const [selectedIndex, setSelectedIndex] = useState(0);
+
+  // Se tiver como usar a tag aqui no lugar de indexToRemove seria melhor
+  // Pq se tiver outro component com o msm numero eles podem ser filtrados ao mesmo tempo
+  const removeTags = indexToRemove => {
+    setTags(tags.filter((_, index) => index !== indexToRemove));
+  };
+
+  const addTags = event => {
+    if (event.key === 'Enter' && event.target.value !== '') {
+      setTags([...tags, event.target.value]);
+      event.target.value = '';
+    }
+  };
+
+  const handleClick = event => {
+    setTags([...tags, event.target.innerText]);
+    event.target.value = '';
+    setIsHidden(!isHidden);
+  };
+
+  const handleFilter = event => {
+    const searchWord = event.target.value;
+    const noRepeatTags = data.filter(e => !tags.includes(e.name));
+    const newFilter = noRepeatTags.filter(value =>
+      value.name.toLowerCase().includes(searchWord.toLowerCase()),
+    );
+    if (searchWord === '') {
+      setFilteredData([]);
+    } else {
+      setIsHidden(false);
+      setFilteredData(newFilter);
+    }
+  };
+
+  return (
+    <div className="container-fluid d-flex flex-column px-0">
+      <div className="d-flex flex-column container-fluid mb-1 px-0">
+        <div className="align-items-start justify-content-start w-100 px-0 position-relative">
+          <input
+            type="text"
+            placeholder={placeholder}
+            className="border border-dark rounded form-control px-2"
+            onKeyUp={e => (e.key === 'Enter' ? addTags(e) : null)}
+            onChange={handleFilter}
```

Comment:
This can make the dropdown disappear when not focusing on the input.
`onBlur={e => setIsHidden(true)}`

Reply:
Thank you, it works. But the problem is, it also hides the tags.